### PR TITLE
Simplify Graql traversals without disjunctions

### PR DESCRIPTION
**Pull request by aelred**
When a traversal contains a single conjunction, avoid the step of union'ing the disjunctions together.

### CODE REVIEW

**Comment by kasper-piskorski** [comment]
does it make any difference or is it just a cleanup thing?

**Comment by kasper-piskorski** [comment]
hmmm rather not a cleanup ;p

**Comment by aelred** [comment]
@kasper-piskorski _should_ have performance implications only :P

File: grakn-graql/src/main/java/ai/grakn/graql/internal/gremlin/GraqlTraversal.java [hunk 1]
```diff
@@ -66,10 +67,18 @@ static GraqlTraversal create(Set<? extends List<Fragment>> fragments) {
...
+        if (fragments().size() == 1) {
+            // If there are no disjunctions, we don't need to union them and get a performance boost
+            return getConjunctionTraversal(graph, graph.admin().getTinkerTraversal().V(), Iterables.getOnlyElement(fragments()));
+        } else {
+            Traversal[] traversals =
+                    fragments().stream().map(list -> getConjunctionTraversal(graph, __.V(), list)).toArray(Traversal[]::new);
+
+            // This is a sneaky trick - we want to do a union but tinkerpop requires all traversals to start from
+            // somewhere, so we start from a single arbitrary vertex.
+            return graph.admin().getTinkerTraversal().V().limit(1).union(traversals);
```
Comment (fppt) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/gremlin/GraqlTraversal.java]: tricky

**Pull request merged**